### PR TITLE
VACMS-15714 Add VAMC System Police content type

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

Relates to #15714
## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

As an admin
1. Go to [/admin/structure/types/manage/vamc_system_va_police/document](https://pr16169-xiwjtl4qmbjrl2kp2oin9dttwrlbc3dt.ci.cms.va.gov/admin/structure/types/manage/vamc_system_va_police/document)
   - [x] Validate that that the content model document exits with links to design, CAIA and others
2. Go to [/node/add/vamc_system_va_police](https://pr16169-xiwjtl4qmbjrl2kp2oin9dttwrlbc3dt.ci.cms.va.gov/node/add/vamc_system_va_police)
3. Complete all required fields (suggest using VISN 20 - VA Alaska to make selection easier)
   - For menu link title use 'VA Police'
   - save the node
   - [x] Validate that the title becomes 'VA Police'
   - [x] Validate the alias is /alaska-health-care/va-police

## Accessibility QA
 As an admin or content editor
For context there are no editor editable fields on this content type.  They will all be set by an install script.
1. Visit https://pr16169-xiwjtl4qmbjrl2kp2oin9dttwrlbc3dt.ci.cms.va.gov/albany-health-care/va-police and evaluate.
2. Edit the node and evaluate.   
- [ ] validate  that all is well or create followup issue(s)

## UX QA
 As an admin or content editor
For context there are no editor editable fields on this content type.  They will all be set by an install script.
1. Visit https://pr16169-xiwjtl4qmbjrl2kp2oin9dttwrlbc3dt.ci.cms.va.gov/albany-health-care/va-police and evaluate.
2. Edit the node and evaluate.   
- [x] validate  that all is well or create followup issue(s)

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
I added perms.

**Comment by laflannery** [comment]
- There is a Preview option at the bottom but I've never seen that in another content type. Is this supposed to be there?
image
- Do the link Text and the Link URL have to be separate, is it possible to combine them like it is on the VAMC System page? Having a naked link like this is bad for accessibility.
**Police page**
image
**VAMC System page**
image

File: config/sync/node.type.vamc_system_va_police.yml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    title_help: ''
+  menu_force:
+    menu_force: true
+    menu_force_parent: false
+name: 'VAMC System VA Police page'
+type: vamc_system_va_police
+description: 'A page about how Veterans see VA Police information for a VAMC system.'
+help: ''
+new_revision: true
+preview_mode: 1
```
Comment (dsasser) [inline config/sync/node.type.vamc_system_va_police.yml:168]: NITPICK:

I know that editors aren't going to be seeing the UI yet (or ever?), but we probably don't need a preview option enabled.

**Review by dsasser — APPROVED** [review]
Looks great to me and QA steps were successful:

Screenshot 2023-11-21 at 10 53 24 AM

I had one minor nitpick but it isn't a blocker.

**Review by edmund-dunn — APPROVED** [review]
All QA steps passed successfully. No issues form me! Nice work!

**Comment by thejordanwood** [comment]
This looks good to me! 

I'm curious to know if the link text/link URL can be combined like @laflannery suggested, but if that isn't possible then this is still approved from a UX perspective.

**Comment by swirtSJW** [comment]
> * There is a Preview option at the bottom but I've never seen that in another content type. Is this supposed to be there?

@dsasser spotted that too.  I will remove it. Good catch.

**Comment by swirtSJW** [comment]
> * Do the link Text and the Link URL have to be separate, is it possible to combine them like it is on the VAMC System page? Having a naked link like this is bad for accessibility.
>   **Police page**

@laflannery and @thejordanwood This is the default rendering for the Spotlight content paragraph.   If we wanted that to render like a normal CTA it would affect everywhere it is used, and would probably be an issue for CMS Platform Team, not product teams.

I believe the original intent of it rendering as separate fields was for proofing the destination separately from the title.  As opposed to preview.  It was a conscious choice that may or may not match our current approach.

File: config/sync/node.type.vamc_system_va_police.yml (hunk 1 above)
Comment (swirtSJW) [inline config/sync/node.type.vamc_system_va_police.yml:168]:
```suggestion
preview_mode: 0
```

Good catch @dsasser Thank you for spotting it.

**Comment by laflannery** [comment]
@swirtSJW I mostly understand but just need to clarify a couple things

> This is the default rendering for the Spotlight content paragraph

1. So this is using the same Spotlight content that VBA/Vet Centers is using?

> it would affect everywhere it is used

2. If #1 is accurate then that's ok, I want it changed everywhere because it's an issue everywhere, I have a [backlog issue for Vet Centers](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10301)

> would probably be an issue for CMS Platform Team, not product teams.

3. I think the POs would need to confirm this because Facilities are the only folks that use Spotlights so they could potentially own these. 
 
Based on the above items, does this change anything or do we still need to just keep this as a separate task to do at a later date?

File: config/sync/core.entity_form_display.node.vamc_system_va_police.default.yml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      show_field_label: 0
+      show_link_to_source: 0
+      show_source_updated_date: 0
+    third_party_settings: {  }
+  field_cc_va_police_overview:
+    type: entity_field_fetch_widget
+    weight: 5
+    region: content
+    settings:
+      show_field_label: 0
```
Comment (davidmpickett) [inline config/sync/core.entity_form_display.node.vamc_system_va_police.default.yml:122]: I would suggest setting this to 1. Helps clarify what the block of text being shown is

**Comment by davidmpickett** [comment]
> @swirtSJW I mostly understand but just need to clarify a couple things
> 
> > This is the default rendering for the Spotlight content paragraph
> 
> 1. So this is using the same Spotlight content that VBA/Vet Centers is using?
> 
> > it would affect everywhere it is used
> 
> 2. If 1 is accurate then that's ok, I want it changed everywhere because it's an issue everywhere, I have a [backlog issue for Vet Centers](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10301)
> 
> > would probably be an issue for CMS Platform Team, not product teams.
> 
> 3. I think the POs would need to confirm this because Facilities are the only folks that use Spotlights so they could potentially own these.
> 
> Based on the above items, does this change anything or do we still need to just keep this as a separate task to do at a later date?

Because of the recursive nature of paragraph types, I think the issue is _actually_ with the **Call To Action** paragraph type (machine name: button) which the **Spotlight content** paragraph type (machine name: featured_content) references. The CTA paragraph is referenced by a bunch of Content Types, most of which are actually Public Websites products

```mermaid
flowchart LR
  nodecampaign_landing_page(("Campaign Landing Page<br>node"))
click nodecampaign_landing_page "/admin/reports/content-model/entity-diagram/node/campaign_landing_page"
  paragraphbutton{{"Call to action<br>paragraph"}}
click paragraphbutton "/admin/reports/content-model/entity-diagram/paragraph/button"
  nodecampaign_landing_page -->|"clp_faq_cta"|paragraphbutton
  nodecampaign_landing_page -->|"clp_resources_cta"|paragraphbutton
  nodecampaign_landing_page -->|"clp_spotlight_cta"|paragraphbutton
  nodecampaign_landing_page -->|"clp_video_panel_more_video"|paragraphbutton
  nodecampaign_landing_page -->|"primary_call_to_action"|paragraphbutton
  nodecampaign_landing_page -->|"secondary_call_to_action"|paragraphbutton
  nodechecklist(("Checklist<br>node"))
click nodechecklist "/admin/reports/content-model/entity-diagram/node/checklist"
  nodechecklist -->|"buttons"|paragraphbutton
  nodefaq_multiple_q_a(("FAQ page<br>node"))
click nodefaq_multiple_q_a "/admin/reports/content-model/entity-diagram/node/faq_multiple_q_a"
  nodefaq_multiple_q_a -->|"buttons"|paragraphbutton
  nodemedia_list_images(("Image list<br>node"))
click nodemedia_list_images "/admin/reports/content-model/entity-diagram/node/media_list_images"
  nodemedia_list_images -->|"buttons"|paragraphbutton
  nodemedia_list_videos(("Video list<br>node"))
click nodemedia_list_videos "/admin/reports/content-model/entity-diagram/node/media_list_videos"
  nodemedia_list_videos -->|"buttons"|paragraphbutton
  nodeq_a(("Reusable Q&A<br>node"))
click nodeq_a "/admin/reports/content-model/entity-diagram/node/q_a"
  nodeq_a -->|"buttons"|paragraphbutton
  nodestep_by_step(("Step-by-Step<br>node"))
click nodestep_by_step "/admin/reports/content-model/entity-diagram/node/step_by_step"
  nodestep_by_step -->|"buttons"|paragraphbutton
  nodesupport_resources_detail_page(("Resources and Support Detail Page<br>node"))
click nodesupport_resources_detail_page "/admin/reports/content-model/entity-diagram/node/support_resources_detail_page"
  nodesupport_resources_detail_page -->|"buttons"|paragraphbutton
  block_contentbenefit_promo>"V2 Home page benefit promo<br>block_content"]
click block_contentbenefit_promo "/admin/reports/content-model/entity-diagram/block_content/benefit_promo"
  block_contentbenefit_promo -->|"promo_cta"|paragraphbutton
  paragraphfeatured_content{{"Spotlight content<br>paragraph"}}
click paragraphfeatured_content "/admin/reports/content-model/entity-diagram/paragraph/featured_content"
  paragraphfeatured_content -->|"cta"|paragraphbutton
style paragraphbutton fill:#ffaacc,stroke:#333,stroke-width:4px;

```

Here's an example of a naked link displaying on a Resources and Support node: resources/your-intent-to-file-a-va-claim

Screenshot 2023-11-27 151402

File: config/sync/auto_entitylabel.settings.node.vamc_system_va_police.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+status: 1
+pattern: 'VA Police'
```
Comment (davidmpickett) [inline config/sync/auto_entitylabel.settings.node.vamc_system_va_police.yml:2]: Per [feedback from CAIA](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68949#issuecomment-1821152958) this should 'VA police'

**Comment by swirtSJW** [comment]
> Something funky going on with Sections.
> 
> When I log in as [david.graves@va.gov](mailto:david.graves@va.gov) who is in the VA Chillicothe health care section, I can edit a VAMC System VA Police Page that is in the Ann Arbor Section, despite not being able to edit any other content in the Ann Arbor section

Good find.   Something is up.  I am looking into it.

**Comment by davidmpickett** [comment]
> > I am pretty sure I know the answer to this, but just to check.
> > Is there an _easy_ way to specify a default "Menu link title" for this content type? We will always want it to be "VA Police" = same as the node title. I couldn't find this as a setting in the Site Building interface and I know menus are weird, but just want to ask and confirm
> 
> @davidmpickett there is not in the content type itself. But we will set them when we bulk create them.
> 
> EDIT: I take that back. We are doing it on other top task pages.

Small note that when this is set it should be 'VA police' lowercase p per copy edits from CAIA

**Comment by swirtSJW** [comment]
> Is there an _easy_ way to specify a default "Menu link title" for this content type? We will always want it to be "VA Police" = same as the node title. I couldn't find this as a setting in the Site Building interface and I know menus are weird, but just want to ask and confirm

@davidmpickett 
I found the issue with this.    The only way I can prefill the menu link title field is if I show the node title (it is disabled, because auto_entity_label provides the value for the title so it is not editable by anyone) .  

image

There is a choice to be made:
A) show a title field that editors can never touch so that menu link title is populated on a page editors will never create themselves
B) Don't show the title which will keep menu link title from being populated on a page that will be built by a script anyway.

What established pattern do we have you ask???
Unfortunately we have both on our top task pages.
image

File: config/sync/core.entity_form_display.node.vamc_system_va_police.default.yml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      show_link_to_source: 0
+      show_source_updated_date: 0
+    third_party_settings: {  }
+  field_cc_term_definitions:
+    type: entity_field_fetch_widget
+    weight: 7
+    region: content
+    settings:
+      show_field_label: 0
```
[the remark this answers is not in the document]
Comment (swirtSJW) [reply config/sync/core.entity_form_display.node.vamc_system_va_police.default.yml]: done

File: config/sync/core.entity_form_display.node.vamc_system_va_police.default.yml (hunk 2 above)
Comment (swirtSJW) [reply config/sync/core.entity_form_display.node.vamc_system_va_police.default.yml]: done

**Comment by swirtSJW** [comment]
The permissions issue has been worked out
image

He can only create content in chilicothe

image

**Comment by swirtSJW** [comment]
image

**Pull request merged**